Composer/packages/client/src/components/AddRemoteSkillModal/CreateSkillModal.tsx

```diff
@@ -0,0 +1,279 @@
+  });
+  const [showIntentSelectDialog, setShowIntentSelectDialog] = useState(false);
+  const [formData, setFormData] = useState<{ manifestUrl: string; endpointName: string }>({
+    manifestUrl: '',
+    endpointName: '',
+  });
+  const [formDataErrors, setFormDataErrors] = useState<SkillFormDataErrors>({});
+  const [skillManifest, setSkillManifest] = useState<any | null>(null);
+  const [showDetail, setShowDetail] = useState(false);
+
+  const { languages, luFeatures, runtime } = useRecoilValue(settingsState(projectId));
+  const { dialogId } = useRecoilValue(designPageLocationState(projectId));
+  const luFiles = useRecoilValue(luFilesSelectorFamily(projectId));
+  const { updateRecognizer } = useRecoilValue(dispatcherState);
+
+  const debouncedValidateManifestURl = useRef(debounce(validateManifestUrl, 500)).current;
+
+  const validationHelpers = {
+    formDataErrors,
+    setFormDataErrors,
+  };
+
+  const options: IDropdownOption[] = useMemo(() => {
+    return skillManifest?.endpoints?.map((item) => {
+      return {
+        key: item.msAppId,
+        text: formatMessage(item.name),
+      };
+    });
+  }, [skillManifest]);
+
+  const handleManifestUrlChange = (_, currentManifestUrl = '') => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { manifestUrl, ...rest } = formData;
+    debouncedValidateManifestURl({
+      formData: { manifestUrl: currentManifestUrl },
+      ...validationHelpers,
+    });
+    setFormData({
+      ...rest,
+      manifestUrl: currentManifestUrl,
+    });
+    setSkillManifest(null);
+  };
+
+  const validateUrl = useCallback(
+    async (event) => {
+      event.preventDefault();
+      setShowDetail(true);
+      try {
+        const { data } = await httpClient.get(`/projects/${projectId}/skill/retrieveSkillManifest`, {
+          params: {
+            url: formData.manifestUrl,
+          },
+        });
+        setSkillManifest(data);
+      } catch (error) {
+        setFormDataErrors({ ...error, manifestUrl: formatMessage('Manifest url can not be accessed') });
+      }
+    },
+    [projectId, formData]
+  );
+
+  const handleSubmit = (event, content: string, enable: boolean) => {
+    event.preventDefault();
+    // add a remote skill, add skill identifier into botProj file
+    addRemoteSkill(formData.manifestUrl, formData.endpointName)
+      .then(() => {
+        TelemetryClient.track('AddNewSkillCompleted');
+        const result = location.href.split('/');
+        let skillId = '';
+        if (result.length > 0) skillId = result[result.length - 1];
+        // add trigger with connect to skill action to root bot
+        const triggerFormData = getTriggerFormData(skillManifest.name, content);
+        addTriggerToRoot(dialogId, triggerFormData, skillId);
+        TelemetryClient.track('AddNewTriggerCompleted', { kind: 'Microsoft.OnIntent' });
+      })
+      .catch((err) => {
+        console.log(err);
```

Comment:
Does this catch need to do something more if the addRemoteSkill failed rather than just logging?